unit_test/solver/network/test_network.py

```diff
@@ -0,0 +1,268 @@
+
+    def test_exists_node(self) -> None:
+        """Test exist node method."""
+        # arrange
+        node = Node(name=uuid.uuid4())
+        self.network.nodes[node.name] = node
+
+        # act
+        result = self.network.exists_node(node_id=node.name)
+
+        # assert
+        self.assertTrue(result)
+
+    def test_get_asset(self) -> None:
+        """Test get asset method."""
+        # arrange
+        self.network.add_existing_asset(asset=self.asset)
+
+        # act
+        result = self.network.get_asset(asset_id=self.asset.name)
+
+        # assert
+        self.assertEqual(result, self.asset)
+
+    def test_get_node(self) -> None:
+        """Test get node method."""
+        # arrange
+        self.network.nodes[self.node.name] = self.node
+
+        # act
+        result = self.network.get_node(node_id=self.node.name)
+
+        # assert
+        self.assertEqual(result, self.node)
+
+    def test_get_asset_error(self) -> None:
+        """Test get asset method raise value error."""
+        # arrange
+
+        # act
+        with self.assertRaises(ValueError) as cm:
+            self.network.get_asset(asset_id=self.asset.name)
+
+        # assert
+        self.assertIsInstance(cm.exception, ValueError)
+        self.assertEqual(str(cm.exception), f"{self.asset.name} Not a valid asset id")
+
+    def test_get_node_error(self) -> None:
+        """Test get node method raise value error."""
+        # arrange
+
+        # act
+        with self.assertRaises(ValueError) as cm:
+            self.network.get_node(node_id=self.node.name)
+
+        # assert
+        self.assertIsInstance(cm.exception, ValueError)
+        self.assertEqual(str(cm.exception), f"{self.node.name} Not a valid node id")
+        pass
+
+    def test_set_result_asset(self) -> None:
+        """Test set result asset method."""
+        # arrange
+        solution = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]
+        asset2 = SolverPipe(name=uuid.uuid4())
+        self.network.add_existing_asset(asset=self.asset)
+        self.network.add_existing_asset(asset=asset2)
+        self.asset.matrix_index = 0
+        asset2.matrix_index = 6
+
+        # act
+        self.network.set_result_asset(solution=solution)  # act
+
+        # assert
+        self.assertEqual(self.asset.prev_sol, solution[:6])
+        self.assertEqual(asset2.prev_sol, solution[6:])
+
+    def test_set_results_node(self) -> None:
+        """Test set result node method."""
+        # arrange
+        solution = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]
+        node = Node(name=uuid.uuid4())
+        self.network.nodes[self.node.name] = self.node
+        self.network.nodes[node.name] = node
+        self.node.matrix_index = 0
+        node.matrix_index = 9
+
+        # act
+        self.network.set_result_node(solution=solution)  # act
+
+        # assert
+        self.assertEqual(self.node.prev_sol, solution[:3])
+        self.assertEqual(node.prev_sol, solution[9:12])
```

Comment:
Zou je ook nog even de print results kunnen testen? Simpele test waarbij we wel weer het test percentag omhoog krijgen :)